### PR TITLE
[MoveosStd] add a function to get signer of module address

### LINKED ISSUE #456: [MovoeosStd] provide a function to get signer of module address
If we provide a native function to get the signer of the module address.

```move
module moveos_std::signer{
   
   #[private_generics(T)]
   public native module_signer<T>(): signer;

}
```

It is safe because we use `private_generics` to restrict the T. 

This function can be used when a developer wants to `move_to` some resource to the module address account. It is also helpful for the resource account.

@steelgeek091  @baichuan3 @wow-sven

### PR DESCRIPTION
**Pull request by bytedeveloperr**
This PR resolves #456

### CODE REVIEW

File: moveos/moveos-stdlib/moveos-stdlib/sources/signer.move [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    assert!(signer_addr == @0x2, 0);
+   }
+
+   #[test]
+   #[expected_failure(abort_code=0, location=moveos_std::signer)]
+   fun test_module_signer_invalid_address() {
+    let signer = module_signer<TestStruct>();
+    let signer_addr = signer::address_of(&signer);
+
+    assert!(signer_addr == @0x6, 0);
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/signer.move:30]: this test case is useless.

**Review by jolestar — APPROVED** [review]

File: moveos/moveos-stdlib/moveos-stdlib/sources/signer.move (hunk 1 above)
Comment (bytedeveloperr) [reply moveos/moveos-stdlib/moveos-stdlib/sources/signer.move]: Okay. I have removed the test case now

**Comment by jolestar** [comment]
Please run

```
cargo run --package rooch-genesis
```

And submit the generated genesis and Move documents.

**Comment by bytedeveloperr** [comment]
Alright, Done 👍

**Pull request merged**